File: frontend/lib/models/transaction_model.dart
class TransactionModel {
  final double amount;
  final String senderEmail;
  final String receiverEmail;
  final String transactionType;
  final String createdAt;

  TransactionModel({
    required this.amount,
    required this.senderEmail,
    required this.receiverEmail,
    required this.transactionType,
    required this.createdAt,
  });

  factory TransactionModel.fromJson(Map<String, dynamic> json) {
    return TransactionModel(
      amount: json['amount'].toDouble(),
      senderEmail: json['sender_email'],
      receiverEmail: json['receiver_email'],
      transactionType: json['transaction_type'],
      createdAt: json['date'],
    );
  }
}

File: banco python/models/transaction_model.py
from mongoengine import Document, StringField, FloatField, DateTimeField
import datetime
from zoneinfo import ZoneInfo

class Transaction(Document):
    sender_email = StringField(required=True)   # Usuario que envía
    receiver_email = StringField(required=True) # Usuario que recibe
    sender_card = StringField(required=True)    # Tarjeta que envía
    receiver_card = StringField(required=True)  # Tarjeta que recibe
    amount = FloatField(required=True)          # Monto
    transaction_type = StringField(default="transfer")
    created_at = DateTimeField(default=lambda: datetime.datetime.now(ZoneInfo('America/Guayaquil'))) # Fecha de transacción
